src/Generation/ResourcesGenerator.php

```diff
@@ -524,45 +524,33 @@ private static function getQueryParams(MethodDetails $method): Vector
      *
      * @return array<string, string> Fields to autopopulate in GAX if unset by user
      */
-    private static function getUuid4FieldsToAutoPopulate(
+    private static function getAutoPopulatedUuid4Fields(
         MethodDetails $method,
-        ServiceDetails $serviceDetails
+        ServiceDetails $service
     ): array {
-        $autoPopulatedFields = Vector::new([]);
-
-        $methodSettings = $serviceDetails->serviceYamlConfig->methodSettings;
-        foreach ($methodSettings as $methodSetting) {
-            $autoPopulatedFieldNames = iterator_to_array($methodSetting->getAutoPopulatedFields());
-            if (!empty($autoPopulatedFieldNames)) {
-
-                // Check if the $methodSetting is for this particular $method
-                $splitMethodName = explode('.', $methodSetting->getSelector());
-                $methodName = $splitMethodName[array_key_last($splitMethodName)];
-                if ($methodName !== $method->name) {
-                    continue;
-                }
+        $result = Map::new();
 
-                // Process only if the method is unary
-                if (!$method->isStreaming()) {
-                    $autoPopulatedFields = $method->optionalFields
-                        ->filter(
-                            // Filter out field names to autopopulate from all fields of the method
-                            fn ($f) => in_array($f->name, $autoPopulatedFieldNames)
-                        )
-                        ->filter(
-                            // Filter for optional field and uuid4 format
-                            fn ($f) => !$f->isRequired && $f->isUuid4
-                        );
-                }
-            }
+        if ($method->isStreaming()) {
+            return [];
+        }
+
+        $methodSetting = $service->serviceYamlConfig->methodSettings
+            ->filter(function ($x) use ($method) {
+                $splitName = explode('.', $x->getSelector());
+                return $method->name == $splitName[array_key_last($splitName)];
+            })->firstOrNull();
+
+        if (!is_null($methodSetting) && count($methodSetting->getAutoPopulatedFields())) {
```

Comment:
Nit: again, perhaps "exit on unfulfilled condition, fallthrough to green path" refactor so

```php
if (is_null(...) || !count(...)) {
  return [];
}

// rest of code 
```

Reply:
Sure. Updated the code.